packages/client/hmi-client/src/page/project/tera-project.vue

```diff
@@ -359,16 +371,42 @@ function openAsset(asset: Tab = tabs.value[activeTabIndex.value], newCode?: stri
 
 	if (newCode) {
 		code.value = newCode;
-		// addCreatedAsset(assetToOpen);
 	}
 }
+
 const openOverview = () =>
 	openAsset({ assetName: 'Overview', assetType: 'overview', assetId: undefined });
 
 function removeClosedTab(tabIndexToRemove: number) {
 	tabStore.removeTab(projectContext.value, tabIndexToRemove);
 }
 
+const updateTabName = (tabName) => {
+	tabs.value[activeTabIndex.value].assetName = tabName;
+};
+
+// Create the new model
+const createNewModel = async (newModel) => {
+	const newModelResp = await createModel(newModel);
+	if (newModelResp) {
+		newAssetId.value = newModelResp.id.toString();
+	}
+
+	if (newAssetId.value) {
+		// add to sidebar
+		await ProjectService.addAsset(props.project.id, ProjectAssetTypes.MODELS, newAssetId.value);
```

Comment:
I wonder if this endpoint return the asset already, I should write an issue about it.

Reply:
It doesn't right now.